internal/fs/dir_handle.go

```diff
@@ -16,6 +16,7 @@ package fs
 
 import (
 	"fmt"
+	"path"
```

Comment:
can you try moving dir_handle.go into handle package and write unit tests.
Having dir_handle in fs package doesn't make sense.